server/src/main/java/org/elasticsearch/index/translog/TranslogHeader.java

```diff
@@ -137,10 +139,17 @@ static TranslogHeader read(final String translogUUID, final Path path, final Fil
             uuid.length = uuidLen;
             in.read(uuid.bytes, uuid.offset, uuid.length);
             // Read the primary term
-            assert version == VERSION_PRIMARY_TERM;
-            final long primaryTerm = in.readLong();
+            final long primaryTerm;
+            if (version == VERSION_PRIMARY_TERM) {
+                primaryTerm = in.readLong();
+            } else {
+                assert version == VERSION_CHECKPOINTS : "Unknown header version [" + version + "]";
```

Comment:
Should we add a comment that this is bwc logic that can be dropped with 5.0?

Reply:
👍 added a comment